### DISCUSSION TITLE
how to run code

### DISCUSSION THREAD

**Discussion by shehzad021** [jdxyw/generativeart#9] [General]
hello brother, this art is really amazing(contour line, noise line and dot line). I'm trying for 2 week to run the code but it's not working. Can you guide how to run and setup files.

**Comment by happyRip** [comment]
It's all documented, try those steps:

- create your project folder and enter that folder `mkdir generativeart && cd generativeart`
- initialize your `go` project with `go mod init generativeart`
- download required packages
  - `go get github.com/jdxyw/generativeart`
  - `go get github.com/jdxyw/generativeart/arts`
- copy any main from examples and save it as `main.go` file:
```go
package main

import (
        "math/rand"
        "time"

        "github.com/jdxyw/generativeart"
        "github.com/jdxyw/generativeart/arts"
        "github.com/jdxyw/generativeart/common"
)

func main() {
        rand.Seed(time.Now().Unix())
        c := generativeart.NewCanva(500, 500)
        c.SetBackground(common.Black)
        c.FillBackground()
        c.SetColorSchema(common.DarkRed)
        c.SetForeground(common.LightPink)
        c.Draw(arts.NewJanus(10, 0.2))
        c.ToPNG("janus.png")
}
```
- run the code with `go run .`
- enjoy your image
![`janus.png`](https://user-images.githubusercontent.com/73070465/155969524-318e7acc-c394-4698-abfa-a91d498fc3fc.png)

**Comment by shehzad021** [comment]
can you explain first steps in more detail. I don't know too much about it

**Reply by happyRip** [reply]
Sure, the question is if you use `windows`, `linux` or `macos`?

**Reply by shehzad021** [reply]
windows

**Reply by happyRip** [reply]
For `windows`:
- open `Windows PowerShell` in search bar
- go to a directory you'd like to have your project stored in, for me it would be `Documents`:
  - `cd Documents`
    - you can specify a full path here, copied from the navigation bar of the `File Explorer`
![image](https://user-images.githubusercontent.com/73070465/155981890-894f013c-0ae2-4913-ad3a-ee70d5317719.png)
and use it as such `cd C:\Users\<your_username>\Documents` (absolute path)
    - some applications (like [Windows Terminal](https://www.microsoft.com/store/productId/9N0DX20HK701)) allow you to 'Open in `...`' when you right-click in a folder
![image](https://user-images.githubusercontent.com/73070465/155982654-53ece6cf-804d-488d-b9aa-8c7e3047b941.png)
- from now on follow my previous instructions:
> - create your project folder and enter that folder `mkdir generativeart && cd generativeart` ...

Bear in mind you need to have `go` installed https://go.dev/doc/install

**Comment by shehzad021** [comment]
please explain initialize step also

**Reply by happyRip** [reply]
`go mod init` is a command that initializes your go project and creates the `go.mod` file, which is needed to download packages with `go get ...` command.

Here is the official documentation for the command https://go.dev/ref/mod#go-mod-init

`go get ...` updates the dependencies of your project. When using `github.com/jdxyw/generativeart` in your project, it becomes a dependency of your program, so you need to `get` it. In my test I had to `get` `github.com/jdxyw/generativeart/arts` as well

Here is the official documentation for the command https://pkg.go.dev/cmd/go#hdr-Add_dependencies_to_current_module_and_install_them

**Comment by shehzad021** [comment]
how can I change colors and save output file (.PNG)

**Reply by jdxyw** [reply]
hey, you could find examples here https://github.com/jdxyw/generativeart/tree/master/example

**Reply by shehzad021** [reply]
and how to save png file

**Reply by jdxyw** [reply]
You could find this line below from the snippet posted above.

`c.ToPNG("janus.png")`

**Reply by shehzad021** [reply]
I want to change color of dot line art

**Reply by jdxyw** [reply]
You can change the color schema from https://github.com/jdxyw/generativeart/blob/master/example/example_dotline.go#L18. You could pass a predefined color schema or pass a customized color array by yourself.

**Reply by shehzad021** [reply]
it shows an error when I write any other colors.
`# command-line-arguments
.\main.go:19:19: undefined: common.Darkblue`

**Comment by shehzad021** [comment]
it shows an error when I write any other colors.
`# command-line-arguments
.\main.go:19:19: undefined: common.Darkblue`

**Reply by jdxyw** [reply]
there is no color schema called `Darkblue`, please check it https://github.com/jdxyw/generativeart/blob/master/common/color.go

**Reply by shehzad021** [reply]
it's still showing error. should I run color file from common

**Reply by happyRip** [reply]
Did you use [PascalCase](https://pl.m.wikipedia.org/wiki/PascalCase)?

**Reply by shehzad021** [reply]
No, how to use it.

**Reply by happyRip** [reply]
Each word begins with a capital letter, so of you have dark blue, you write `DarkBlue`, but please check if the colors you'd like to try are present in the file sent above

**Reply by shehzad021** [reply]
it didn't not I tried colors from the file like
`LightPink`

**Reply by jdxyw** [reply]
have you used the Go before?  Can you run those examples under example folders?

**Reply by shehzad021** [reply]
all other examples are working with new colors define in common file but only dot line is not working.
